### ISSUE TITLE
Missing scalar uniform hints in visual shaders

### ISSUE THREAD

**Issue by dweremeichik** [godotengine/godot#55571]
### Godot version

v3.4.stable.official [206ba70f4]

### System information

N/A

### Issue description

When writing shaders by hand, or converting from Spatial Materials, scalar uniform hints can be used. An example might be:

```
uniform float roughness : hint_range(0,1);
uniform float point_size : hint_range(0,128);
```

However when using visual shaders, there is no way to specify uniform hints.

### Steps to reproduce

1. Create a visual shader. 
2. Create a scalar uniform in the visual shader. 
3. View shader params - no hint available.

**Comment by Chaosus** [comment]
This was implemented in 4.0, which seems needed for porting back.

**Comment by Chaosus** [comment]
Fixed in https://github.com/godotengine/godot/pull/56466
